core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -747,6 +747,15 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         listing.synchronized {
           listing.delete(classOf[LogInfo], rootPath.toString)
         }
+      case _: FileNotFoundException
+          if reader.rootPath.getName.endsWith(EventLogFileWriter.IN_PROGRESS) =>
+        val finalFileName = reader.rootPath.getName.stripSuffix(EventLogFileWriter.IN_PROGRESS)
+        if (fs.exists(new Path(reader.rootPath.getParent, finalFileName))) {
+          // Do nothing, the application completed during processing, the final event log file
+          // will be processed by next around.
+        } else {
+          logError(s"File does not exist: ${reader.rootPath}")
```

Comment:
Could you revise this message to include both `IN_PROGRESS` and the final location?

Reply:
In addition, this is not a critical error in SHS. `logWarning` could be enough in this case.

Reply:
Thanks, updated